engine/access/rest/converter.go

```diff
@@ -3,21 +3,201 @@ package rest
+	for _, arg := range tx.Arguments {
+		// todo validate
+		args = append(args, []byte(arg))
+	}
+
+	proposal, err := toProposalKey(tx.ProposalKey)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	payer, err := toAddress(tx.Payer)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	auths := make([]flow.Address, len(tx.Authorizers))
+	for _, auth := range tx.Authorizers {
+		a, err := toAddress(auth)
+		if err != nil {
+			return flow.TransactionBody{}, err
+		}
+
+		auths = append(auths, a)
+	}
+
+	payloadSigs, err := toTransactionSignatures(tx.PayloadSignatures)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	envelopeSigs, err := toTransactionSignatures(tx.EnvelopeSignatures)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	return flow.TransactionBody{
+		ReferenceBlockID:   flow.Identifier{},
+		Script:             []byte(tx.Script),
+		Arguments:          args,
+		GasLimit:           uint64(tx.GasLimit),
+		ProposalKey:        proposal,
+		Payer:              payer,
+		Authorizers:        auths,
+		PayloadSignatures:  payloadSigs,
+		EnvelopeSignatures: envelopeSigs,
+	}, nil
+}
+
+// Response section - converting flow models to response models.
+
+func proposalKeyResponse(key *flow.ProposalKey) *generated.ProposalKey {
+	return &generated.ProposalKey{
+		Address:        key.Address.String(),
+		KeyIndex:       int32(key.KeyIndex),
+		SequenceNumber: int32(key.SequenceNumber),
+	}
+}
+
+func transactionSignatureResponse(signatures []flow.TransactionSignature) []generated.TransactionSignature {
+	sigs := make([]generated.TransactionSignature, len(signatures))
+	for _, sig := range signatures {
+		sigs = append(sigs,
+			generated.TransactionSignature{
+				Address:     sig.Address.String(),
+				SignerIndex: int32(sig.SignerIndex),
+				KeyIndex:    int32(sig.KeyIndex),
+				Signature:   string(sig.Signature),
+			},
+		)
+	}
+
+	return sigs
+}
+
+func transactionResponse(tx *flow.TransactionBody) *generated.Transaction {
+	var args []string
+	for _, arg := range tx.Arguments {
+		args = append(args, string(arg))
+	}
+
+	var auths []string
+	for _, auth := range tx.Authorizers {
+		auths = append(auths, auth.String())
+	}
+
+	return &generated.Transaction{
+		Id:                 tx.ID().String(),
+		Script:             string(tx.Script),
+		Arguments:          args,
+		ReferenceBlockId:   tx.ReferenceBlockID.String(),
+		GasLimit:           int32(tx.GasLimit), // todo(sideninja) make sure this is ok
+		Payer:              tx.Payer.String(),
+		ProposalKey:        proposalKeyResponse(&tx.ProposalKey),
+		Authorizers:        auths,
+		PayloadSignatures:  transactionSignatureResponse(tx.PayloadSignatures),
+		EnvelopeSignatures: transactionSignatureResponse(tx.EnvelopeSignatures),
+		Result:             nil, // todo(sideninja) should we provide result, maybe have a wait for result http long pulling system would be super helpful (with reasonable timeout) but careful about resources and dos
```

Comment:
Yes, also the result should only actually be included if it is `expand`ed, because I think according to the spec, this field is not `expand`ed by default.

Reply:
The expanded and select is being worked upon in this PR - https://github.com/onflow/flow-go/pull/1645/files. It is based off a branch which is based off of this branch :)